nndownload/nndownload.py

```diff
@@ -2005,170 +2005,134 @@ def download_video_comments(
+            thread.start()
+            tasks.append({
+                "thread": thread,
+            })
+
+        for task in tasks:
+            task["thread"].join()
+
+        comments_data["globalComments"]["retrievedCount"] = sum(thread["retrievedCount"] for thread in comments_data["threads"])
+        comments_data["globalComments"]["commentCount"] = template_params["comment_count"]
 
-    except KeyboardInterrupt:
-        output("\nCTRL+C detected. Saving downloaded comments for {0}\n".format(template_params["id"]), logging.INFO)
-        raise
-    finally:
-        # Save in case of success and on interrupt)
         with open(filename, "w", encoding="utf-8") as file:
-            json.dump(COMMENTS_DATA_JSON, file, indent=None, ensure_ascii=False, sort_keys=True)
+            json.dump(comments_data, file, indent=None, ensure_ascii=False, sort_keys=True)
+
+        output("Saved comments for {0} to {1}.\n".format(template_params["id"], filename), logging.INFO)
+
 
-def fetch_thread_comments(
+def fetch_comments_thread(
     session: requests.Session,
     video_id: str,
     api_server: str,
     thread_key: str,
     thread: dict,
     language: str,
-    last_time: int,
-    comments_limit: int,
-    thread_data: dict
+    progress: Progress,
+    comments_data: dict,
+    comments_from: int,
+    comments_limit: int = None,
 ) -> None:
-    """Fetch comments for thread (requires login)."""
-    base_data = {
-        "threadKey": thread_key,
-        "params": {
-            "language": language,
-            "targets": [thread]
-        }
+    """Fetch comments for a specific comments thread."""
+
+    thread_data = {
+        "comments": [],
+        "fork": thread["fork"],
+        "id": thread["id"]
     }
-    fetched_comments_count = 0
-    previous_comments = None
-    total_comments = None
-
-    with Progress(
-        TextColumn("[progress.description]{task.description}"),
-        BarColumn(),
-        TaskProgressColumn(),
-        TextColumn("({task.completed}/{task.total})"),
-        TimeRemainingColumn(),
-        transient=True
-    ) as progress:
-        task = progress.add_task(f"Downloading {thread['fork']} comments", total=None)
-        
-        while fetched_comments_count < comments_limit:
-            try:
-                payload = {
-                    **base_data,
-                    "additionals": {
-                        "res_from": -1000,
-                        "when": last_time
-                    }
-                }
-                
-                headers = {
-                    **API_HEADERS,
-                    "content-type": "text/plain;charset=UTF-8",
-                    "x-client-os-type": "others",
+
+    task_id = progress.add_task(thread["fork"], total=None, visible=False)
+    while not progress.tasks[task_id].finished:
```

Comment:
Setting the bars `visible=False`, we can first fetch the total, then update them and set them visible after. Using checks on the task itself like `progress.tasks[task_id].finished` helps us prevent having to track things ourself with a bunch of local variables